extensions/interactions/PencilCodeEditor/PencilCodeEditor.js

```diff
@@ -69,7 +69,26 @@ oppia.directive('oppiaInteractivePencilCodeEditor', [
           });
 
           $scope.reset = function() {
-            pce.setCode($scope.initialCode);
+            var resetConfirmed = $modal.open({
```

Comment:
I'm not sure about the name here. It sounds like a boolean.

Do you even need to name it? You can do:

```
    $modal.open({
      ...
    }).result.then(function(result) {
       ...
    });
```